xtask/src/main.rs

```diff
@@ -0,0 +1,126 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+use anyhow::{bail, Result};
+use clap::Parser;
+use std::collections::BTreeMap;
+use std::fmt::Write;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+
+#[derive(Debug, Parser)]
+enum Xtask {
+    /// Packages bootleby into a hubris compatible archive
+    Package {
+        #[clap(long)]
+        board: String,
+        #[clap(long)]
+        out: PathBuf,
+    },
+}
+
+// borrowed from hubris
+fn remap_paths() -> Result<BTreeMap<PathBuf, &'static str>> {
+    // Panic messages in crates have a long prefix; we'll shorten it using
+    // the --remap-path-prefix argument to reduce message size.  We'll remap
+    // local (Hubris) crates to /hubris, crates.io to /crates.io, and git
+    // dependencies to /git
+    let mut remap_paths = BTreeMap::new();
+
+    if let Ok(home) = std::env::var("CARGO_HOME") {
+        let cargo_home = PathBuf::from(home);
+        let cargo_git = cargo_home.join("git").join("checkouts");
+        remap_paths.insert(cargo_git, "/git");
+
+        // This hash is canonical-ish: Cargo tries hard not to change it
+        // https://github.com/rust-lang/cargo/blob/5dfdd59/src/cargo/core/source_id.rs#L755-L794
+        //
+        // It depends on system architecture, so this won't work on (for example)
+        // a Raspberry Pi, but the only downside is that panic messages will
+        // be longer.
+        let cargo_registry = cargo_home
+            .join("registry")
+            .join("src")
+            .join("github.com-1ecc6299db9ec823");
+        remap_paths.insert(cargo_registry, "/crates.io");
+        // If Cargo uses the sparse registry (stabilized since ~1.72) it caches fetched crates
+        // in a slightly different path. Remap that one as well.
+        //
+        // This path has the same canonical-ish properties as above.
+        let cargo_sparse_registry = cargo_home
+            .join("registry")
+            .join("src")
+            .join("index.crates.io-6f17d22bba15001f");
+        remap_paths.insert(cargo_sparse_registry, "/crates.io");
+    }
+
+    if let Ok(dir) = std::env::var("CARGO_MANIFEST_DIR") {
+        let mut hubris_dir = std::path::PathBuf::from(dir);
+        hubris_dir.pop();
+        hubris_dir.pop();
+        remap_paths.insert(hubris_dir.to_path_buf(), "/bootleby");
```

Comment:
```suggestion
        let mut bootleby_dir = std::path::PathBuf::from(dir);
        bootleby_dir.pop();
        remap_paths.insert(bootleby_dir.to_path_buf(), "/bootleby");
```
(`CARGO_MANIFEST_DIR` points to `bootleby/xtask`, versus `hubris/build/xtask`, so we don't need as many pops)